From 3dafc4c508e4928d459719958918052ce49fd223 Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Tue, 12 Aug 2014 15:56:08 +0200
Subject: [PATCH] totemnet: Add totemnet_member_set_active

totemnet_member_set_active together with transport specific
member_set_active makes possible for totemnet (and more interestingly
transport) to be informed about membership changes.

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Christine Caulfield <ccaulfie@redhat.com>
---
 exec/totemnet.c |   23 +++++++++++++++++++++++
 exec/totemnet.h |    5 +++++
 2 files changed, 28 insertions(+), 0 deletions(-)

diff --git a/exec/totemnet.c b/exec/totemnet.c
index 2786423..82846f9 100644
--- a/exec/totemnet.c
+++ b/exec/totemnet.c
@@ -122,6 +122,11 @@ struct transport {
 	int (*member_remove) (
 		void *transport_context,
 		const struct totem_ip_address *member);
+
+	int (*member_set_active) (
+		void *transport_context,
+		const struct totem_ip_address *member,
+		int active);
 };
 
 struct transport transport_entries[] = {
@@ -460,3 +465,21 @@ extern int totemnet_member_remove (
 
 	return (res);
 }
+
+int totemnet_member_set_active (
+	void *net_context,
+	const struct totem_ip_address *member,
+	int active)
+{
+	struct totemnet_instance *instance = (struct totemnet_instance *)net_context;
+	unsigned int res = 0;
+
+	if (instance->transport->member_set_active) {
+		res = instance->transport->member_set_active (
+			instance->transport_context,
+			member,
+			active);
+	}
+
+	return (res);
+}
diff --git a/exec/totemnet.h b/exec/totemnet.h
index c5b8126..288dabb 100644
--- a/exec/totemnet.h
+++ b/exec/totemnet.h
@@ -124,4 +124,9 @@ extern int totemnet_member_remove (
 	void *net_context,
 	const struct totem_ip_address *member);
 
+extern int totemnet_member_set_active (
+	void *net_context,
+	const struct totem_ip_address *member,
+	int active);
+
 #endif /* TOTEMNET_H_DEFINED */
-- 
1.7.1

